include/glow/Base/Tensor.h

```diff
@@ -82,6 +147,48 @@ class Tensor final {
   char *getData() const { return data_; }
 
 public:
+  /// \returns true if tensor data is stored on a device
+  bool isDeviceResident() const { return residencyInfoP_->isDeviceResident(); }
+
+  // Update device residency info with new device manager and context
+  void moveToDevice(DeviceTensorTransferManager *deviceManager,
+                    void *locationContext) {
+    residencyInfoP_->deviceManager_ = deviceManager;
```

Comment:
Nit: Is it performance critical to keep the implementations of this and many other device residency-related methods in the header file so that it can be inlined? If not, I'd suggest moving them into a cpp implementation file. `Tensor.h` is used virtually everywhere and changing anything in it results in recompiling pretty-much everything. Thus, anything that can be kept out of it without any negative performance implications, should be better kept out of it.